synapse/storage/databases/main/presence.py

```diff
@@ -434,13 +434,21 @@ def _get_active_presence(self, db_conn: Connection) -> List[UserPresenceState]:
 
         txn = db_conn.cursor()
         txn.execute(sql, (PresenceState.OFFLINE,))
-        rows = self.db_pool.cursor_to_dict(txn)
+        rows = txn.fetchall()
         txn.close()
 
-        for row in rows:
-            row["currently_active"] = bool(row["currently_active"])
-
-        return [UserPresenceState(**row) for row in rows]
+        return [
+            UserPresenceState(
+                user_id=user_id,
+                state=state,
+                last_active_ts=last_active_ts,
+                last_federation_update_ts=last_federation_update_ts,
+                last_user_sync_ts=last_user_sync_ts,
+                status_msg=status_msg,
+                currently_active=bool(currently_active),
+            )
+            for user_id, state, last_active_ts, last_federation_update_ts, last_user_sync_ts, status_msg, currently_active in rows
+        ]
```

Comment:
>  (My guess is you'd save a few from not making and resizing the rows list, but otherwise not much.)

Yeah I'm not sure it is much safer; I was a bit scared about the `txn.close()` so tried to keep the code as close as possible.